### PR TITLE
test: add pipline related test

### PR DESCRIPTION
**Pull request by Vinkle-hzt**
here is the test list:
- transport test
- pipeline example test

### CODE REVIEW

File: .github/workflows/test_on_dev.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Additional Test on Dev
```
Comment (yezhem) [inline .github/workflows/test_on_dev.yml:1]: can we just add the test case to "run_on_dev.yml"??

**Review by yezhem — CHANGES_REQUESTED** [review]

**Review by yezhem — CHANGES_REQUESTED** [review]
the xxx_test.py will be run by pytest, so we don't need to add the test cast to ci workflow.
if you want to add it to ci workflow, change the file's name

**Comment by Vinkle-hzt** [comment]
The code has been modified according to comments.

File: .github/workflows/run_on_dev.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: Test LLM Models on Dev
+name: Additional Test on Dev
```
Comment (yezhem) [inline .github/workflows/run_on_dev.yml:1]: Additional Test on Dev -> Test on Dev

File: .github/workflows/run_on_dev.yml [hunk 3]
```diff
@@ -24,6 +24,7 @@ jobs:
       - name: install dependencies
         run: |
           cd /workspace/multi-lora-fine-tune
+          pip install -e .
```
Comment (yezhem) [inline .github/workflows/run_on_dev.yml:27]: this is necessary?

File: tests/pipeline_example_check.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        test_grads = pool.starmap(
+            pipe_process,
+            args)
+
+    res_1, res_2 = test_grads
+
+    grads_1, weight, datas = res_1
+    grads_2, _, _ = res_2
+
+    # create a model with the same weight
```
Comment (yezhem) [inline tests/pipeline_example_check.py:31]: split this function to test_by_pipe and test_by_seq.

Comment (yezhem) [reply tests/pipeline_example_check.py:31]: or add more comment in it.

File: tests/transport_check.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    for i in range(loop_times):
+        message = PipeMessage(
+            src_=src,
+            dst_=dst,
+            msg_id_=i,
+            msg_type_=task_type,
+            tensor_data_=torch.ones((4096, 4096)).cuda(0) * i,
+            batch_data_=None,
+        )
+        msgs.append(message)
```
Comment (yezhem) [inline tests/transport_check.py:26]: why use the msgs?

**Review by yezhem — CHANGES_REQUESTED** [review]

File: .github/workflows/run_on_dev.yml (hunk 3 above)
Comment (Vinkle-hzt) [reply .github/workflows/run_on_dev.yml:27]: `pipeline_example_check.py` and `transport_check.py` will use mlora module, so we should make a link first.

File: .github/workflows/run_on_dev.yml [hunk 6]
```diff
@@ -7,7 +7,7 @@ on:
         - 'main'
 
 jobs:
-  test-llama-7b:
+  additonal-test:
```
Comment (yezhem) [inline .github/workflows/run_on_dev.yml:10]: additonal-test -> test ??

**Review by yezhem — APPROVED** [review]
LGTM

**Pull request merged**